### PR TITLE
[docs] add documentation on how to make a Pr with Gitpod.

### LINKED ISSUE #1253: Gitpod should clone a fork
> Extracted from https://github.com/iterative/dvc.org/pull/1226#issuecomment-624418719 and https://github.com/iterative/dvc.org/pull/1226#issuecomment-624437520

I tried using Gitpod from https://dvc.org/doc/user-guide/contributing/docs#development-environment to submit a change. It was all working nicely, (I even installed a VS Code plugin to the IDE.) But after committing with Git, I'm unable to push because I didn't grant Gitpod permission to push directly to dvc.org

image

Open source contributors will never have this permission. Can it be made so a fork is created in their accounts first, and then Gitpod launched from there?

---

For example, I tried with another account I own, @AyokVoluntarios, which doesn't have push access to this repo, but it also cloned the upstream directly:

image

So again, even if I tell Github to let Gitpod push to my repos, I simply can't do it from that account.

image

The user would need to configure the Git remote and branch manually from the IDE. Do you know whether that's the only solution, @nisarhassan12? Thanks

### PR DESCRIPTION
**Pull request by nisarhassan12**
Closes #1253

### CODE REVIEW

File: content/docs/user-guide/contributing/docs.md [hunk 1]
```diff
@@ -53,11 +53,45 @@ We will review your PR as soon as possible. Thank you for contributing!
...
+> it'll automatically clone the repo, install the dependencies, and run the app
+> (takes a few minutes), so that you can start straight away.
+>
+> <a href="https://gitpod.io/#https://github.com/iterative/dvc.org">
+>   <img src="https://gitpod.io/button/open-in-gitpod.svg" alt="Open in Gitpod">
+> </a>
+
+<details>
+
+### How to make a Pull Request with Gitpod?
```
Comment (shcheklein) [inline content/docs/user-guide/contributing/docs.md:71]: @jorgeorpinel @nisarhassan12 feels it's not a good place for very long instructions like this in this doc, sorry. It's a good thing to mention that there is a possibility, may be provide the button, but docs like this should be maintained but the Gitpod, not us.

Comment (nisarhassan12) [reply content/docs/user-guide/contributing/docs.md:71]: @shcheklein I agree. Making Prs with Gitpod is briefly documented here https://www.gitpod.io/docs/pull-requests/ should I update the Pr to only have the link instead ?

Comment (shcheklein) [reply content/docs/user-guide/contributing/docs.md:71]: @nisarhassan12 I'll let @jorgeorpinel decide on this :)

Comment (nisarhassan12) [reply content/docs/user-guide/contributing/docs.md:71]: ok. Thanks.

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:71]: Hmm I thought the expandable section would be appropriate for this. I would just put it inside the block quote. But indeed we shouldn't have to maintain Gitpod instructions that may change in the future... Sorry, I guess I didn't think it through. Let me think... ⏳

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:71]: I guess we could just link to https://www.gitpod.io/docs/git/and https://www.gitpod.io/docs/pull-requests/ but the forking part is not mentioned anywhere unfortunately, which is my main concern.

We cold also have a separate user-guide like we have on some 3rd party tools e.g. https://dvc.org/doc/user-guide/setup-google-drive-remote or https://dvc.org/doc/user-guide/running-dvc-on-windows

I supose it'll be easier for me to take over this PR. Thanks for your contributions so far Nisar! (I hope you enabled repo maintainers to push to the PR branch 🙂)

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:71]: p.s. please notice my feature request in your repo @nisarhassan12 thanks: gitpod-io/gitpod/issues/1482

**Comment by jorgeorpinel** [comment]
Preview: https://dvc-landing-master-re3jtk2lrzp.herokuapp.com/doc/user-guide/contributing/docs#development-environment

File: content/docs/user-guide/contributing/docs.md [hunk 2]
```diff
@@ -53,11 +53,45 @@ We will review your PR as soon as possible. Thank you for contributing!
...
+
+- Fork the repo by selecting Find Command from View in the top menu and then
+  type fork and press enter:
+
+  ![image](https://user-images.githubusercontent.com/46004116/76392493-75727880-6393-11ea-8939-41e6c36477de.png)
```
Comment (shcheklein) [inline content/docs/user-guide/contributing/docs.md:76]: We should host these images on the dvc.org (e.g. to process them properly with sharp)? not sure what are the consequences of sharing it like this. Btw, @nisarhassan12 how did you create them?

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:76]: Maybe they're somewhere in this repo: https://github.com/gitpod-io/workspace-images

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:76]: Ah no, I think he pasted them into a Github conversation like this:

image
> `image`

Comment (jorgeorpinel) [reply content/docs/user-guide/contributing/docs.md:76]: Which is interesting. I wonder it Github ever deletes images from it's CDN.

Comment (nisarhassan12) [reply content/docs/user-guide/contributing/docs.md:76]: yeah! True I pasted them in Github comment. Sorry, this is a bad idea.

File: content/docs/user-guide/contributing/docs.md (hunk 1 above)
Comment (nisarhassan12) [reply content/docs/user-guide/contributing/docs.md:71]: Ok. Thanks.

Comment (nisarhassan12) [reply content/docs/user-guide/contributing/docs.md:71]: > p.s. please notice my feature request in your repo @nisarhassan12 thanks: gitpod-io/gitpod/issues/1482

@jorgeorpinel I'm not sure about this I think I have not unchecked the following while making the PR:

image

Please let me know if you are not able to pus to my branch.

**Comment by shcheklein** [comment]
@nisarhassan12 @jorgeorpinel any updates on this one?

**Comment by nisarhassan12** [comment]
> I supose it'll be easier for me to take over this PR. Thanks for your contributions so far Nisar! (I hope you enabled repo maintainers to push to the PR branch :slightly_smiling_face:)

@shcheklein Thanks. @jorgeorpinel said the above :point_up: in https://github.com/iterative/dvc.org/pull/1262#discussion_r422308466

**Comment by shcheklein** [comment]
@nisarhassan12 I see, got it. Thanks.

**Comment by jorgeorpinel** [comment]
Yes, I still have this under my radar!

**Comment by shcheklein** [comment]
Closing this for now. After a few discussions with the team I think we should keep the docs more or less general, and avoid providing external links unless they provide some different flow, bring substantially different, faster, etc. experience.

Some thoughts behind this logic. In our case we have two types of ad-hoc contributors:

1. Editors - they don't need this for sure. They can edit MD files with Github.
2. Front-end developers - they should be already aware about the tools the prefer to use (editors, etc). If they prefer online editor - up to them, it's still a regular flow with `yarn`, etc.

Also, we def should avoid using Github URLs for assets, or complicated instructions.

**Comment by jorgeorpinel** [comment]
Understood. I do like Gitpod as well as https://github.com/features/codespaces but not sure whether we should mention them somewhere or how. Maybe in the README.md ... Will think about it.

**Pull request closed without merging**